app/screens/CameraScreen.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import * as Permissions from 'expo-permissions'
+import { Camera } from 'expo-camera'
+import styled from 'styled-components/native'
+import { Dimensions, Text, View } from 'react-native'
+import Container from '../components/Container'
+import Header from '../components/Header'
+import getPlatformStyles from '../utils/get-platform-styles'
+import Icon from '../components/Collecticons'
+import { colors } from '../style/variables'
+import { savePhoto } from '../actions/camera'
+import * as Location from 'expo-location'
+import LoadingOverlay from '../components/LoadingOverlay'
+
+const win = Dimensions.get('window')
+const buttonStyles = getPlatformStyles({
+  ios: {
+    bottom: 100
+  },
+  iphoneX: {
+    bottom: 100
+  },
+  android: {
+    bottom: 20
+  }
+})
+
+const SnapButton = styled.TouchableHighlight`
+  position: absolute;
+  border-radius: ${Math.round(win.width + win.height) / 2};
+  width: 60;
+  height: 60;
+  background-color: ${colors.primary};
+  right: 40%;
+  bottom: ${buttonStyles.bottom};
+  justify-content: center;
+  align-items: center;
+  shadow-color: grey;
+  shadow-opacity: 0.7;
+  shadow-offset: 0px 0px;
+`
+
+class CameraScreen extends React.Component {
+  static navigationOptions = ({ navigation }) => {
+    return {
+      title: 'Camera'
+    }
+  }
+  state = {
+    hasCameraPermission: null,
+    type: Camera.Constants.Type.back,
+    saving: false
+  }
+
+  async componentWillMount () {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA)
+    this.setState({
+      hasCameraPermission: status === 'granted'
+    })
+  }
+
+  async snap () {
+    if (this.camera) {
+      this.setState({
+        saving: true
+      })
+      let { uri, width, height } = await this.camera.takePictureAsync()
+      const location = await Location.getCurrentPositionAsync({})
+      console.log(uri, width, height, location)
```

Comment:
remove console.log?

Reply:
Thinking I'll keep this here for a bit, it's kind of the only way to say the camera took a picture right now.